### PR TITLE
GTJ-27 login frontend

### PR DESCRIPTION
**Pull request by zvorsi**

### CODE REVIEW

File: backend/src/controllers/sessionController.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ export const sessionController = {
   async post(req, res) {
     try {
       const data = await sessionService.login(req.body);
+      console.log(data);
```
Comment (adamgyulavari) [inline backend/src/controllers/sessionController.js:7]: @zvorsi remove console log

File: backend/src/services/sessionService.js [hunk 2]
```diff
@@ -11,9 +11,11 @@ export const sessionService = {
     if (username && password) {
       try {
         const result = await userService.get({ username, password });
+        console.log(result);
```
Comment (adamgyulavari) [inline backend/src/services/sessionService.js:14]: @zvorsi remove console.log

File: frontend/src/App.js [hunk 3]
```diff
@@ -5,17 +5,17 @@ import './App.css';
...
 
 function App() {
-  localStorage.setItem(
-    'TRIBES_TOKEN',
-    'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySUQiOjMsImlhdCI6MTU5ODA0MTk3Mn0.cVAcR4UiobKN05oicJYQYPIkibMws4yIATb8GVuVD_g'
-  );
+  // localStorage.setItem(
+  //   'TRIBES_TOKEN',
+  //   'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySUQiOjMsImlhdCI6MTU5ODA0MTk3Mn0.cVAcR4UiobKN05oicJYQYPIkibMws4yIATb8GVuVD_g'
+  // );
```
Comment (adamgyulavari) [inline frontend/src/App.js:18]: @zvorsi remove comments

File: frontend/src/actions/LoginActions.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import {
+  SET_KINGDOM_SUCCESS,
+  SET_ERROR_SUCCESS
+  } from '../constants/ActionTypes';
+
+export const LoginActions = (data) => {
+    const {username, password} = data;
```
Comment (adamgyulavari) [inline frontend/src/actions/LoginActions.js:7]: @zvorsi destructure in the parameter instead

File: frontend/src/actions/LoginActions.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    return dispatch => {
+      dispatch({
+        type: 'kiscica',
+      });
```
Comment (adamgyulavari) [inline frontend/src/actions/LoginActions.js:11]: @zvorsi dispatch a meaningful action if dispatching anything eg `START_LOGIN`

File: frontend/src/components/login/Login.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import './Login.css';
+import { LoginActions } from '../../actions/LoginActions';
+
+
+class Login extends Component{
```
Comment (adamgyulavari) [inline frontend/src/components/login/Login.js:8]: @zvorsi styling
```suggestion
class Login extends Component {
```

File: frontend/src/components/login/Login.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      password: '',
+    };
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleUsername = this.handleUsername.bind(this);
+    this.handlePassword = this.handlePassword.bind(this);
+    // this.login = this.login(this);
```
Comment (adamgyulavari) [inline frontend/src/components/login/Login.js:19]: @zvorsi remove comment

File: frontend/src/components/login/Login.js [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    let userData = this.state;
+    const { history } = this.props;
+
+    console.log(userData);
```
Comment (adamgyulavari) [inline frontend/src/components/login/Login.js:28]: @zvorsi remove console log

File: frontend/src/actions/LoginActions.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
+import {
+  SET_KINGDOM_SUCCESS,
+  SET_ERROR_SUCCESS
+  } from '../constants/ActionTypes';
+
+export const LoginActions = (data) => {
```
Comment (adamgyulavari) [inline frontend/src/actions/LoginActions.js:6]: @zvorsi this the `loginAction` not plural

File: frontend/src/components/login/Login.js [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+   }
+
+    handleUsername(e) {
+      let value = e.target.value;
+      this.setState(prevState => ({ ...prevState, username: value }));
+    }
+
+    handlePassword(e) {
+      let value = e.target.value;
+      this.setState(prevState => ({ ...prevState, password: value }));
```
Comment (adamgyulavari) [inline frontend/src/components/login/Login.js:43]: @zvorsi unnecessary value variable
```suggestion
      this.setState(prevState => ({ ...prevState, password: e.target.value }));
```

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

**Comment by adamgyulavari** [comment]
@zvorsi also match the PR name with the others

File: frontend/src/components/login/Login.js (hunk 10 above)
Comment (zvorsi) [reply frontend/src/components/login/Login.js:43]: tried, but immediately fails when trying to login, I'd rather leave it be

File: frontend/src/actions/LoginAction.js [hunk 11]
```diff
@@ -3,13 +3,12 @@ import {
   SET_ERROR_SUCCESS
   } from '../constants/ActionTypes';
 
-export const LoginActions = (data) => {
-    const {username, password} = data;
+export const loginAction = ({username, password}) => {
+    // const {username, password} = data;
```
Comment (adamgyulavari) [inline frontend/src/actions/LoginAction.js:7]: @zvorsi 🤦‍♂️

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

**Review by adamgyulavari — APPROVED** [review]

**Pull request merged**